### DISCUSSION TITLE
Header showPerPage

### DISCUSSION THREAD

**Discussion by akulmehta** [Power-Components/livewire-powergrid#520] [Ideas]
## Feature Request
Ability to `showPerPage` in header.

### Summary

Would like to have the ability to show the per page field on the top of the table in the header rather than the footer.

### Why is this needed?

Having the ability to toggle per page entries at the top allows the user to change this value without going to the very bottom of the page. This becomes very important when the size of the table becomes larger as the user would have to scroll down significantly.

### Suggested Solution(s)

Would like to have the show per page options to be visible at the top of the table rather than the bottom.

### Related issue(s)/PR(s)

None I could find.

**Comment by luanfreitasdev** [comment]
You can currently use this custom functionality

https://livewire-powergrid.com/#/table/features-setup?id=includeviewontop

**Reply by akulmehta** [reply]
@luanfreitasdev could you provide more details on how to do this? Currently the `showperPage` is displayed as a part of the `footer.blade.php`. Should I extract the part for the show per page i.e.:

```blade
                @if(data_get($setUp, 'footer.perPage') && count(data_get($setUp, 'footer.perPageValues')) > 1)
                    <div class="d-flex flex-lg-row align-items-center">
                        <label class="w-auto">
                            <select wire:model.lazy="setUp.footer.perPage" class="form-select">
                                @foreach(data_get($setUp, 'footer.perPageValues') as $value)
                                    <option value="{{$value}}">
                                        @if($value == 0)
                                            {{ trans('livewire-powergrid::datatable.labels.all') }}
                                        @else {{ $value }}
                                        @endif
                                    </option>
                                @endforeach
                            </select>
                        </label>
                        <small class="ms-2 text-muted">
                            {{ trans('livewire-powergrid::datatable.labels.results_per_page') }}
                        </small>
                    </div>
                @endif
```
into a separate component and use it?

**Reply by luanfreitasdev** [reply]
Hello, @akulmehta , it would be exactly that

**Reply by akulmehta** [reply]
@luanfreitasdev Hmm tried it - doesn't work. 
![image](https://user-images.githubusercontent.com/42812604/181363256-58952aa1-815d-4da3-b030-cb8ec82c825a.png)
and
![image](https://user-images.githubusercontent.com/42812604/181363300-34457694-f148-49ff-8a7e-9a2bd27a658e.png)

even the `@dump` doesn't show up.

**Reply by akulmehta** [reply]
@luanfreitasdev any ideas what I am doing wrong?

**Comment by akulmehta** [comment]
@luanfreitasdev I figured out the issue. The `includeViewOnTop` method requires the path to the view in dot notation as it uses `@includeIf` directive. I have created a PR to update the docs to reflect the correct notation at https://github.com/Power-Components/powergrid-doc/pull/39 - please consider merging.

**Reply by luanfreitasdev** [reply]
Thank you very much for the observation!.
